IanGRap/Squish
# Respawn tweaks 1

Contributors: Coulter, Ian

Goal: Make various tweaks to the way respawn blocks work, with the high-level goal being to make them more powerful and more impactful during matches. Also Adding placeholder SFX and UI indicator.

Files Changed:
- Spawner.cs: Added a respawn function specifically for respawn blocks.
- BlockInput.cs: Repurposed old script to allow players to move their respawn blocks using thumbstick inputs.
- RespawnUI.cs: New script that spawns the UI image for each respawn block and destroys the image when the block appears on the screen.
- PlayerRespawner.cs: New script that respawns the player when they kill someone as a block.
- Respawn Canvas: New canvas that spawns and holds the UI images as its children.
- Respawn Image: A skelegoo icon + an arrow pointing upward, spawned by the Canvas with the color of the corresponding respawn block.
- BlockCollisionDelegator.cs: Fixed a bug that allowed players to push long and tall blocks upward.
- Respawn Beam: Made the beam a child of the respawning player on instantiate(), moved the transform to avoid breaking blocks below the respawning player.

How to test: 
- Have a player die. Their respawn block should spawn in immediately, with a UI indicator at the top of the screen that tracks them until they enter the camera view.
- Move the block using the left thumbstick. It should be able to move and push other blocks in all four directions, but cannot break itself against the environment. Can be pushed and broken by players. It should also no longer break once its velocity is zero (i.e. reaching the ground or landing on a stationary block)
- Use the respawn block to kill someone. The controlling player should instantly respawn at the location of the kill, with a beam that destroys all blocks directly above the respawned player.

Notes: The code for this one is probably in need of some refactoring and triage, so any suggestions for improving it are welcome. Ideas for improving the UI are also welcome. Shout-out to Ian for helping me work through getting the UI set up. Also, there is a known bug that I discovered and decided to leave out of this PR, wherein two players who die simultaneously will have their respawn blocks spawn inside one another.
